### DISCUSSION TITLE
How to use smart_quotes correctly?

### DISCUSSION THREAD

**Discussion by dmke** [gettalong/kramdown#804] [Q&A]
Hi,

I'm having trouble to enable smart-quote parsing/conversion.

<details><summary>Gemfile</summary>

```ruby
source "https://rubygems.org"
ruby "~> 3.0.0"
gem "kramdown"
```

</details>
<details><summary>test.rb</summary>

```ruby
#!/usr/bin/env ruby

require "bundler"
Bundler.require

p({ ruby: RUBY_VERSION, kramdown: Kramdown::VERSION })

S = %("Hi", she said. "Hrm'wa?" he grunted back.)

[
  {},                                            # nothing
  { smart_quotes: "‚‘„“".codepoints },           # unicode codepoints
  { smart_quotes: %w[‚ ‘ „ “] },                 # unicode array
  { smart_quotes: "sbquo,lsquo,bdquo,ldquo" },   # entities, comma sep
  { smart_quotes: "sbquo,lsquo,bdquo,ldquo" },   # entities, space sep
  { smart_quotes: %w[sbquo lsquo bdquo ldquo] }, # entities, array
].each do |sq|
  [
    {},
    { typographic_symbols: { sbquo: "‚", lsquo: "‘", bdquo: "„", ldquo: "“" } },
  ].each do |ts|
    %w[html markdown].each do |input|
      opts = { input: input }.merge(sq, ts)

      s = input == "markdown" ? S : "<p>#{S}</p>"
      puts Kramdown::Document.new(s, opts).to_latex.strip
    end
  end
end
```

</details>
<details open><summary>output</summary>

```bash
$ ruby test.rb
{:ruby=>"3.0.6", :kd=>"2.4.0"}
"Hi", she said. "Hrm'wa?" he grunted back.
[repeated output omitted]
```

</details>

I would have expected to get at least one output with proper typographic quotes, like

```
„Hi“, she said. „Hrm‘wa?“ he grunted back.
```

Changing `.to_latex` to `.to_kramdown` changes the output to

```
\"Hi\", she said. \"Hrm\'wa?\" he grunted back.
```

i.e. Kramdown seems to detect the quotation marks correctly, but doesn't apply the transformation.

Am I missing an option to enable them?

**Comment by gettalong** [comment]
In the shell:

~~~
$ echo '"Hi", she said. "Hrm'"'"'wa?" he grunted back.' | kramdown --smart-quotes "sbquo,lsquo,bdquo,ldquo"
<p>„Hi“, she said. „Hrm‘wa?“ he grunted back.</p>
$ echo '"Hi", she said. "Hrm'"'"'wa?" he grunted back.' | kramdown --smart-quotes "sbquo,lsquo,bdquo,ldquo" --input markdown
<p>"Hi", she said. "Hrm'wa?" he grunted back.</p>
~~~

You need to use the kramdown parser if you want smart quotes as the markdown parser doesn't support them (see https://github.com/gettalong/kramdown/blob/master/lib/kramdown/parser/markdown.rb#L30).

**Reply by dmke** [reply]
I should clarify: I have HTML input and want LaTeX output.

I guess I need to subclass `Kramdown::Converter::Latex` and refine `#escape`?

**Reply by gettalong** [reply]
Hmm... I don't think I can provide you with a quick solution for that combination.

**Reply by dmke** [reply]
Don't worry, for my needs I think I can adopt code I've written [elsewhere](https://github.com/digineo/texd-ruby/blob/master/lib/texd/helpers.rb).
